### ISSUE TITLE
Returning training history as a Pandas DataFrame

### ISSUE THREAD

**Issue by kl2792** [pytorchbearer/torchbearer#372]
Currently what I'm doing is:

``` python
history = trial.run(epochs=5, verbose=0) # using Jupyter-compatible printer
pd.DataFrame.from_records(dict(v, train_samples=k[0], val_samples=k[1])
                          for k, v in history)
```

<img width="934" alt="screen shot 2018-08-26 at 4 01 25 pm" src="https://user-images.githubusercontent.com/7944969/44632495-724a9f80-a949-11e8-8bbc-1ed7d70ce08f.png">

Another plus of the DataFrame method is that the Trial object can order metrics (e.g. loss comes before accuracy, etc.), and it can include information in addition to metrics (such as time taken per epoch, etc.).

What are your thoughts on adding Pandas as a dependency and having this changed? I'm happy to put in a PR with the code above. I also understand if you don't want to add Pandas as a dependency.

**Comment by ethanwharris** [comment]
I like the idea of being able to do more with history. Ideally, the history would be an object with methods for getting data frames, numpy matrices etc. That way only the to_pandas method (or whatever it would be called) would need to import pandas, so it would only need to be a dependency in requirements.txt not setup.py. I'll have a look at doing something like this

**Comment by ethanwharris** [comment]
Finally got round to coming up with a solution for this. There is now an example showing the recommended way to get a data frame from history. We've simply updated the format of history so that it is always just a list of records, this means that a data frame can be made with a simple one-liner and it is more obvious what the training / validation steps are. I hope that helps!
